sdk/src/main/java/io/opentelemetry/sdk/metrics/export/IntervalMetricReader.java

```diff
@@ -202,11 +206,15 @@ private Exporter(InternalState internalState) {
 
     @Override
     public void run() {
-      List<MetricData> metricsList = new ArrayList<>();
-      for (MetricProducer metricProducer : internalState.getMetricProducers()) {
-        metricsList.addAll(metricProducer.getAllMetrics());
+      try {
+        List<MetricData> metricsList = new ArrayList<>();
+        for (MetricProducer metricProducer : internalState.getMetricProducers()) {
```

Comment:
Do we need this section to be in this `try` block too?

Reply:
Perhaps not, but if anything in here fails, we don't want the IMR to die, so it felt safer just to put everything in the block.